Sming/Services/WebHelpers/escape.cpp

```diff
@@ -219,16 +159,21 @@ char *uri_unescape(char *dest, size_t dest_len, const char *src, int src_len)
 /* calculates the required length for html_escape */
 unsigned html_escape_len(const char *s, size_t len) {
 	unsigned ret;
-	for(ret=0;len && *s;len--,s++) {
-		switch(*s) {
-			case '<': ret+=4; break; /* &lt; */
-			case '>': ret+=4; break; /* &gt; */
-			case '&': ret+=5; break; /* &amp; */
-			case '"': ret+=6; break; /* &quot; */
-			case '\'': ret+=6; break; /* &apos; */
-			default:
-				ret++;
-		}
+	for(ret=0;len && *s;len--, s++) {
+		char c = *s;
```

Comment:
What is the advantage of replacing the switch statement with if statement?

Reply:
With embedded stuff I tend to use if because the resulting code is more predictable. In this case I probably did it because it makes the code a little more readable (less verbose). I doubt it will affect code performance one way or the other here.